src/freenet/client/filter/CSSTokenizerFilter.java

```diff
@@ -1835,6 +1851,7 @@ else if(HTMLelement.indexOf('#')!=-1)
 		            if(!((c>='a' && c<='z') || (c>='A' && c<='Z')))
 		                return null;
 		            for(int i=1;i<attSelectionParts[0].length();i++) {
+		                c=attSelectionParts[0].charAt(i);
```

Comment:
I’m not sure what this does but it sure looks like something a test should catch. 🙂

Reply:
The  fixed test actually caught that ☺

`CSS2_BAD_SELECTOR.add("h1[foo,=bar] {}");` ← if we had had this one before, this would have been caught by the test.